SteamKit2/SteamKit2/Base/Generated/SteamMsgWorkshop.cs

```diff
@@ -77,9 +77,31 @@ public bool needs_action
 
     }
 
-    public interface IWorkshop
+    public class Workshop : SteamUnifiedMessages.IUnifiedService
     {
-        CWorkshop_GetEULAStatus_Response GetEULAStatus(CWorkshop_GetEULAStatus_Request request);
+        public static string ServiceName { get; } = "Workshop";
+
+        /// <inheritdoc />
+        public SteamUnifiedMessages UnifiedMessages { get; init; }
```

Comment:
What if a consumer wants to generate and use their own service?

Reply:
Good point. I think an explicit interface implementation can solve for that: https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/interfaces/explicit-interface-implementation

It would hide access to the property on the implementation types and only allow access if you cast to a `IUnifiedService`, which the `SteamUnifiedMessages` handler can easily do.

A consumer could end up doing the same, but it would at least make it more unlikely.

Reply:
Yeah it would be good to get rid of this property from the generated classes, and only leave it on the parent.